src/system-update-generator/system-update-generator.c

```diff
@@ -18,21 +19,23 @@
 static const char *arg_dest = NULL;
 
 static int generate_symlink(void) {
-        const char *p = NULL;
-
-        if (laccess("/system-update", F_OK) < 0) {
-                if (errno == ENOENT)
-                        return 0;
-
-                log_error_errno(errno, "Failed to check for system update: %m");
-                return -EINVAL;
+        const char *fn;
+        FOREACH_STRING(fn, "/system-update", "/etc/system-update") {
+                int r = laccess(fn, F_OK);
+                if (r == 0) {
+                        const char *p = strjoina(arg_dest, "/" SPECIAL_DEFAULT_TARGET);
```

Comment:
strjoina() is built on alloca(), and stack allocation is not OK to be called in a loop. Consider splitting out the fag file finding into its own function (i.e. `system_update_flag_exists()`), which would avoid the problem and make this more readable and more similar to the old version...